FlowCal/plot.py

```diff
@@ -1058,21 +1141,33 @@ def scatter2d(data_list,
         plt.ylabel(data_plot.channels[1])
 
     # Set scale of axes
-    plt.gca().set_xscale(xscale)
-    plt.gca().set_yscale(yscale)
+    plt.gca().set_xscale(xscale, data=data_list, channel=channels[0])
+    plt.gca().set_yscale(yscale, data=data_list, channel=channels[1])
 
-    # Set plot limits if specified, else extract range from data_plot
+    # Set plot limits if specified, else extract range from data_list
     # Use data_plot.hist_bins with one single bin
-    if xlim is not None:
-        plt.xlim(xlim)
-    elif hasattr(data_plot, 'hist_bins') and \
-            hasattr(data_plot.hist_bins, '__call__'):
-        plt.xlim(data_plot.hist_bins(channels=0, nbins=1, scale=xscale))
-    if ylim is not None:
-        plt.ylim(ylim)
-    elif hasattr(data_plot, 'hist_bins') and \
-            hasattr(data_plot.hist_bins, '__call__'):
-        plt.ylim(data_plot.hist_bins(channels=1, nbins=1, scale=yscale))
+    if xlim is None:
+        xlim = [np.inf, -np.inf]
+        for data in data_list:
+            if hasattr(data, 'hist_bins') and \
+                    hasattr(data.hist_bins, '__call__'):
+                xlim_data = data.hist_bins(channels=channels[0],
+                                           nbins=1,
+                                           scale=xscale)
```

Comment:
I guess I feel like I'm OK with cleverly generating 1 large hist bin via `hist_bins()` to get the lower limit. Would just be nice to answer my initial question in the code via a comment, though (because this particular implementation really does beg that question).

I think setting the range to start at 1 would be cleaner for this particular use case (less code, less explanation). Not sure if that's a good idea everywhere, though. Don't remember all the places where range is used or why exactly we went with 0 as the bottom in the first place. If we went with `[1,$PnR]`, would the 0 bin get put with the 1 bin for integer data and $PnR bins generated (i.e. a full resolution histogram for integer data)? And we don't care about data less than 0 because we have logicle display now? Does that data just get thrown out if you manually specify log? (I think you should still be able to plot on log scales with negative data have it either gracefully fail or plot some intuitive subset).

Ultimately I think I'll defer to what you think is best here.